### ISSUE TITLE
Wrong warning when a saved model (from TF2.4) is loaded using tf-nightly 

### ISSUE THREAD

**Issue by jvishnuvardhan** [tensorflow/tensorflow#46385]
**System information**
- Have I written custom code (as opposed to using a stock example script provided in TensorFlow):
- OS Platform and Distribution (e.g., Linux Ubuntu 16.04):
- Mobile device (e.g. iPhone 8, Pixel 2, Samsung Galaxy) if the issue happens on mobile device:
- TensorFlow installed from (source or binary): pip
- TensorFlow version (use command below): TF2.4, tf-nightly
- Python version:
- Bazel version (if compiling from source):
- GCC/Compiler version (if compiling from source):
- CUDA/cuDNN version:
- GPU model and memory:

**Describe the current behavior**
Using TF2.4, a Keras model was saved using `model.save`.  Later, when I tried to load the saved model using  `tf-nightly`, code throws unexpected error as shown below.

```
2.5.0-dev20210112
WARNING:tensorflow:SavedModel saved prior to TF 2.4 detected when loading Keras model. Please ensure that you are saving the model with model.save() or tf.keras.models.save_model(), *NOT* tf.saved_model.save(). To confirm, there should be a file named "keras_metadata.pb" in the SavedModel directory.
WARNING:tensorflow:No training configuration found in save file, so the model was *not* compiled. Compile it manually.
```
 

**Describe the expected behavior**

The following warning should be revised as the initial model was saved using `TF 2.4` but warning complains that the model was saved prior to TF 2.4.

`WARNING:tensorflow:SavedModel saved prior to TF 2.4 detected when loading Keras model. Please ensure that you are saving the model with model.save() or tf.keras.models.save_model(), *NOT* tf.saved_model.save(). To confirm, there should be a file named "keras_metadata.pb" in the SavedModel directory.`

The above warning was noticed only when the model was saved in `tf` format. The above warning doesn't appear when the model was saved in `h5` format.

**Standalone code to reproduce the issue**

Please check the [gist here](https://colab.research.google.com/gist/jvishnuvardhan/e3ec0d7b92034affc4b7644a89dc7ebc/untitled483.ipynb).

**Other info / logs** Include any logs or source code that would be helpful to
diagnose the problem. If including tracebacks, please include the full
traceback. Large logs and files should be attached.
This issue was originally opened earlier [here](https://github.com/tensorflow/tensorflow/issues/45288)

**Comment by ravikyram** [comment]
@jvishnuvardhan 

I have tried in colab with saving the model in TF 2.4 and loading the model in TF nightly version(`2.5.0-dev20210113`)  was able to reproduce the issue.Please, find the gist [here](https://colab.research.google.com/gist/ravikyram/d5f00f44515e0bbd1faf3dd2f5c51ae5/untitled612.ipynb).

However If i save and load the model in TF nightly version (`2.5.0-dev20210113`) gist [here](https://colab.research.google.com/gist/ravikyram/fde17b37206baf586914a488d847c946/untitled613.ipynb) or If i ave the model and load the model in TF 2.4 version gist [here](https://colab.research.google.com/gist/ravikyram/a68d1374495aeaad9c64a51b1b22e44f/untitled614.ipynb) then i am not seeing the below error message .
`WARNING:tensorflow:SavedModel saved prior to TF 2.4 detected when loading Keras model. Please ensure that you are saving the model with model.save() or tf.keras.models.save_model(), *NOT* tf.saved_model.save(). To confirm, there should be a file named "keras_metadata.pb" in the SavedModel directory.`
Thanks!

**Comment by jvishnuvardhan** [comment]
This is intended behavior and the warning is to encourage users to use `model.save`  instead of `tf.saved_model.save` when saving keras models. 

[Here](https://colab.research.google.com/gist/jvishnuvardhan/53116f96ec3127dab87c3b109d355846/untitled94.ipynb) is a gist where model was trained and saved using `TF2.4`. Then, loaded the model using `tf-nightly` and trained again on the same data. The accuracy with `TF2.4` and `tf-nightly` are similar which shows that there is no problem loading the model using `tf-nightly`. Thanks!

**Comment by SysuJayce** [comment]
Hi @jvishnuvardhan , there is no `keras_metadata.pb` in tf2.4 saved_model dir. But in tf2.5, I found `keras_metadata.pb`.

Do you know what's the point of `keras_metadata.pb`?
Is it compatible with prior tf version that has no `keras_metadata.pb`?

**Comment by apivovarov** [comment]
In TF 2.5.0 `model.save("model_dir")` creates one additional file in the model folder - `keras_metadata.pb`.
The difference btw prev versions is that now we can load the model back and get fully-featured Keras model.

```
model=tf.keras.models.load_model("model_dir")
type(model)
# tensorflow.python.keras.engine.sequential.Sequential

model._is_graph_network
# True

model.summary() # Works!
```
We can even re-fit the model!
We can save it in h5 format (some tools, such as Netron, support only h5 format)

**Comment by SysuJayce** [comment]
> In TF 2.5.0 `model.save("model_dir")` creates one additional file in the model folder - `keras_metadata.pb`.
> The difference btw prev versions is that now we can load the model back and get fully-featured Keras model.
> 
> ```
> model=tf.keras.models.load_model("model_dir")
> type(model)
> # tensorflow.python.keras.engine.sequential.Sequential
> 
> model._is_graph_network
> # True
> 
> model.summary() # Works!
> ```
> 
> We can even re-fit the model!
> We can save it in h5 format (some tools, such as Netron, support only h5 format)

@apivovarov Hi, thanks for the reply.

So if we use `load_model()` in tf2.5, the states of optimizer will be restored automatically.
However, we need to use `custom_objects` in tf2.4 if we have used some custom layers.

And if we want to deploy the saved model in tf2.5, we can simply delete the `keras_metadata.pb` to reduce the file size, and everything will remain the same as tf2.4.

Please correct me if there is any mistakes!

**Comment by legel** [comment]
Suppose I am past the point of no return and I cannot follow the "encouragement" of this warning. How do I suppress it?
